File: backend/GrivanveAnalyticDotNetMicroServiceCode/GrievanceAnalyticsService/DTOs/GrievanceStatsDTO.cs
//This DTO (Data Transfer Object) is used only for sending custom/statistical data to the frontend
//not for storing in the DB
namespace GrievanceAnalyticsService.DTOs
{
    public class GrievanceStatsDTO
    {       
        public int TotalGrievances { get; set; }
        public int Open { get; set; }
        public int Pending { get; set; }
        public int Resolved { get; set; }
    }
}

File: backend/src/main/java/com/app/dto/GrievanceStatsDTO.java
package com.app.dto;



import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;

@Data // Generates getters, setters, toString, equals, hashCode
@Schema(description = "DTO for grievance statistics returned from .NET analytics service")
public class GrievanceStatsDTO {

    @Schema(description = "Total number of grievances synced", example = "5")
    private int totalGrievances;

    @Schema(description = "Number of grievances with status OPEN", example = "2")
    private int open;

    @Schema(description = "Number of grievances with status PENDING", example = "1")
    private int pending;

    @Schema(description = "Number of grievances with status RESOLVED", example = "2")
    private int resolved;

}